apps/server/src/services/ElasticSearchClient.ts

```diff
@@ -278,3 +276,51 @@ export async function getRecommendedInstitutions(
 
   return institutions
 }
+
+export async function deleteRemovedInstitutions(
+  newInstitutions: CachedInstitution[]
+) {
+  const newInsIdsSet = new Set(
+    newInstitutions.map((ins: CachedInstitution) => ins.ucp_id)
+  )
+
+  const oldInstitutions = JSON.parse(
```

Comment:
IMO we should be looking at the cache for oldInstitutions. This list is going to be outdated, and that is going to result in the elastic search documents being out of sync.

Reply:
Hmm I was basically treating the `ucwInstitutionMapping.json` file as a cache, I guess that's why originally I was updating that file each time the widget poller triggered an update.

The elastic search docker container is being initialized from the json file every time you stop and start the server, so the elastic search institutions always start out in sync with that file until polling happens and then elastic search syncs with the institution server. 

We could use redis for cache as well, but it probably will introduce some complexity. It could maybe work if on startup we check redis first and if it contains an institution list then load ElasticSearch via redis but if redis doesn't have the list then we load from the json file. That would make it so editing the institution list manually wouldn't do anything unless you delete the redis cache.

Reply:
If we made it so ElasticSearch data persists between restarts then we might not have to worry about these issues and we could plug the oldInstitutions ucp_ids into redis to be used for checking if any institutions need to be deleted from ElasticSearch

Reply:
I made Redis in a docker container so it will reset on every restart and should stay in sync with Elasticsearch